reverse_file_buffered.py

```diff
@@ -0,0 +1,36 @@
+from file_io import Io, FileIo
+
+
+def reverse_file_in_memory(io: Io, name: str, buf_size: int):
+    """Reverse a content of the file `name` using no more than buf_size of additional memory"""
+    assert buf_size >= 2, "Minimum allowed buf_size is 2"
+
+    length = io.file_length(name)
+    byte_size = buf_size//2
```

Comment:
let's keep var names up to date -- this value is definitely not a byte_size. If you process some part at a time, it is commonly named "chunk". So "chunk_size" may be a good option